### ISSUE TITLE
Netdata packaging in case of binary distribution

### ISSUE THREAD

**Issue by gamefundas** [netdata/netdata#7135]
OS: CentOS Amazon Linux
Netdata version: 1.18.1

I am a bit puzzled around the installed folder structure.

Install command used:
./netdata-v1.18.1.gz.run --accept

This resulted in the following structure

[ec2-user@stage-springboot-app netdata]$ cd /opt/netdata/
[ec2-user@stage-springboot-app netdata]$ ls -ltr
total 32
drwxr-xr-x 3 netdata netdata 4096 Oct 18 14:46 lib
drwxrwxr-x 6 netdata netdata 4096 Oct 18 14:46 share
drwxr-xr-x 3 netdata netdata 4096 Oct 18 14:46 include
drwxr-xr-x 3 netdata netdata 4096 Oct 18 14:47 etc
drwxr-xr-x 6 netdata netdata 4096 Oct 18 14:47 var
drwxrwxr-x 2 netdata netdata 4096 Oct 18 14:47 system
drwxrwxr-x 3 netdata netdata 4096 Oct 18 14:47 bin
drwxrwxr-x 5 netdata netdata 4096 Oct 19 01:21 usr
lrwxrwxrwx 1 netdata netdata    3 Oct 19 01:21 sbin -> bin
lrwxrwxrwx 1 netdata netdata   21 Oct 19 01:21 netdata-web-files -> usr/share/netdata/web
lrwxrwxrwx 1 netdata netdata   11 Oct 19 01:21 netdata-configs -> etc/netdata
lrwxrwxrwx 1 netdata netdata   19 Oct 19 01:21 netdata-plugins -> usr/libexec/netdata
lrwxrwxrwx 1 netdata netdata   17 Oct 19 01:21 netdata-metrics -> var/cache/netdata
lrwxrwxrwx 1 netdata netdata   15 Oct 19 01:21 netdata-logs -> var/log/netdata
lrwxrwxrwx 1 netdata netdata   15 Oct 19 01:21 netdata-dbs -> var/lib/netdata

These symlinks look strange to me as I believe these "netdata-*" should actually map to the system folders. For example I would have imagined the /opt/netdata/etc/netdata should map to the "/etc/netdata" folder.

Currently the /etc/netdata is empty, even if I copy my netdata.conf and stream.conf to this folder, Netdata doesnt seem to be picking it up as it refers to "/opt/netdata/etc/netdata".

**Comment by thiagoftsm** [comment]
Hi @gamefundas ,

The specific  file that you got creates a kind of `chroot` at `/opt`, this is the motive you are seeing shared links pointing directly to `/opt/netdata/`
Case you wanna have the files direct in the normal file system directories, you will need to use the `netdata-installer.sh` to compile or the RPM files we release, for more details you can check our documentation https://docs.netdata.cloud/packaging/installer/.

Best regards!

**Comment by gamefundas** [comment]
I am getting the following errors for EL6 RPM install. Looks like these dependencies arent accessible from my local yum repo.

--> Finished Dependency Resolution
Error: Package: netdata-1.18.1-1.el6.i686 (/netdata-1.18.1-1.el6.i686)
           Requires: Judy
Error: Package: netdata-1.18.1-1.el6.i686 (/netdata-1.18.1-1.el6.i686)
           Requires: protobuf-c
Error: Package: netdata-1.18.1-1.el6.i686 (/netdata-1.18.1-1.el6.i686)
           Requires: libuv >= 1
Error: Package: netdata-1.18.1-1.el6.i686 (/netdata-1.18.1-1.el6.i686)
           Requires: freeipmi
Error: Package: netdata-1.18.1-1.el6.i686 (/netdata-1.18.1-1.el6.i686)
           Requires: libipmimonitoring.so.5
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

**Comment by thiagoftsm** [comment]
I am assuming you followed the instructions that are present on https://packagecloud.io/netdata/netdata/install#bash-rpm
I never tested an i686 on Centos, but on x86_64 I am getting these packages from the following repositories:

![VirtualBox_Centos_19_10_2019_20_04_45](https://user-images.githubusercontent.com/49162938/67150794-b6b2a080-f2ab-11e9-8c55-2b084de503fd.png)

Please, pay attention for the fact that pgdg-redhat-all is not necessary for Netdata. Another thing important to say that I showed to you an output from a Virtual Machine it is not Amazon server.

**Comment by gamefundas** [comment]
Tried x86_64 as well and it didnt work. I may have to give up on this. Its not working with the Amazon CentOs linux distribution. Building from binary lead to some gcc compilation issues so thats not working either. Tried the docker image and it has an issue recognizing any containers. Opened another bug related to that.

Only option is to use the binary and change all my installation scripts to point to the new folder structure.

**Comment by thiagoftsm** [comment]
@gamefundas I also assigned this issue for our DevOps team for they have knowledge about this problem you had.
Please, can you send us the compilation messages for we to have condition to adjust our installer?

**Comment by gamefundas** [comment]
The compilation dependency error has to do with a bad gcc installation on the Amazon linux image so its perhaps not a Netdata issue. Shuffling few different dependencies I was able to install gcc and then Netdata builds fine.

I eventually modified my AMI to use the Netdata binary installations. So guess I am good for now. Not sure if this binary will work with various flavors of Amazon ec2 types. Will test along the way and see how this turns out.

We should be able to close this issue for now.
